azure/durable_functions/decorators/decorators.py

```diff
@@ -0,0 +1,139 @@
+from .triggers import OrchestrationTrigger, ActivityTrigger, EntityTrigger, EntityClient, OrchestrationClient, DurableClient
+from typing import Callable, Dict, List, Optional
+from azure.durable_functions.entity import Entity
+from azure.durable_functions.orchestrator import Orchestrator
+from typing import Union
+from azure.functions import FunctionRegister, TriggerApi, BindingApi, AuthLevel
+
+class DurableFunctionApp(FunctionRegister, TriggerApi, BindingApi):
+
+    def __init__(self,
+                http_auth_level: Union[AuthLevel, str] = AuthLevel.FUNCTION):
+        """Constructor of :class:`FunctionApp` object.
+        :param http_auth_level: Determines what keys, if any, need to be
+        present
+        on the request in order to invoke the function.
+        """
+        super().__init__(auth_level=http_auth_level)
+
+    def _configure_entity_callable(self, wrap) -> Callable:
+        """Decorator function on user defined function to create and return
+            :class:`FunctionBuilder` object from :class:`Callable` func.
+        """
+
+        def decorator(entity_func):
+            handle = Entity.create(entity_func)
+            handle.__name__ = entity_func.__name__
+
+            return wrap(handle)
+
+        return decorator
+
+    def _configure_orchestrator_callable(self, wrap) -> Callable:
+        """Decorator function on user defined function to create and return
+            :class:`FunctionBuilder` object from :class:`Callable` func.
+        """
+
+        def decorator(orchestrator_func):
+            handle = Orchestrator.create(orchestrator_func)
+            handle.__name__ = orchestrator_func.__name__
+
+            return wrap(handle)
+
+        return decorator
+
+    def on_orchestration_change(self, name: str,
+                                orchestration: Optional[str] = None):
+        @self._configure_orchestrator_callable
```

Comment:
nit: add comments for all the decorator methods

Reply:
done!